src/Deku/Interpret.js

```diff
@@ -326,23 +500,58 @@ export const makeRoot_ = (a) => (state) => () => {
 	};
 };
 
-export const giveNewParent_ = (a) => (state) => () => {
-	if (state.units[a.id]) {
-
-		var ptr = a.id;
-		var parent = a.parent;
+export const giveNewParent_ = (runOnJust) => (b) => (state) => () => {
+	const runMe = []
+	if (state.units[b.id] && state.units[b.id].startBeacon) {
+		var c = b;
```

Comment:
does this need to be `state.units[b.id].startBeacon`? not sure what the logic is here

Reply:
or does `b` come immediately after `state.units[b.id].startBeacon`?

Reply:
The logic here is that if we're giving a new parent to something, there's one of two possibilities:

1. The element is a plain ol' element, in which case there's no `startBeacon` and we can add it as-is.
2. The element is a `dyn`, in which case we need to give a new parent to everything between the start beacon and the end beacon inclusive. So `runMe` should contain all of those nodes.

I'll write a test that verifies this logic, it's currently untested!

Reply:
@MonoidMusician added <https://github.com/mikesol/purescript-deku/blob/6cb66f71f1c3553cbb3f154d61a47999e5e18857/test/Test/Main.purs#L133>, which is the same as the test above it except that instead of inserting a `span`, it inserts a `fixed` with two spans and then moves it up on the page. The test asserts that all of the elements in a fragment get shifted upwards correctly.